### PR TITLE
Change RSpace reset to skip resetting to already selected root

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
It is wasteful to reset to the same root in RSpace if it was not changed.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
umbrella ticket: https://rchain.atlassian.net/browse/RCHAIN-3060 

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: node/src/test/scala/coop/rchain/node/configuration/commandline/Ed25519PubKeyConverterSpec.scala [hunk 1]
```diff
@@ -13,9 +13,9 @@ class Ed25519PubKeyConverterSpec extends PropSpec with GeneratorDrivenPropertyCh
     val gen = Gen.listOfN(Ed25519.publicKeyLength, Arbitrary.arbByte.arbitrary).map(_.toArray)
 
     forAll(gen) { bytes: Array[Byte] =>
-      val asHex           = Base16.encode(bytes)
+      val asHex = Base16.encode(bytes)
```
Comment (lukasz-golebiewski) [inline node/src/test/scala/coop/rchain/node/configuration/commandline/Ed25519PubKeyConverterSpec.scala]: Having those formatting related changes in virtually every PR is getting annoying

**Comment by lukasz-golebiewski** [comment]
Is the correct behaviour tested anywhere?

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by woky** [comment]
bors try

**Pull request merged**